tests/migration_test.go

```diff
@@ -721,6 +723,103 @@ var _ = Describe("[rfe_id:393][crit:high][vendor:cnv-qe@redhat.com][level:system
+						}
+					}
+
+					return false, nil
+				})
+				if err != nil {
+					return "", err
+				}
+
+				return vmiIP, nil
+			}
+
+			waitForPodCompleted := func(podNamespace string, podName string) error {
+				pod, err := virtClient.CoreV1().Pods(podNamespace).Get(context.TODO(), podName, k8smetav1.GetOptions{})
+				if err != nil {
+					return err
+				}
+				if pod.Status.Phase == k8sv1.PodSucceeded || pod.Status.Phase == k8sv1.PodFailed {
+					return nil
+				}
+				return fmt.Errorf("pod hasn't completed, current Phase: %s", pod.Status.Phase)
+			}
+
+			BeforeEach(func() {
+				clientVMI = tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling))
+				clientVMI.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(fedoraVMSize)
+				clientVMI.Spec.Domain.Devices.Interfaces = []v1.Interface{
+					{
+						Name: defaultNetworkName,
+						InterfaceBindingMethod: v1.InterfaceBindingMethod{
+							Bridge: &v1.InterfaceBridge{},
+						},
+					},
+				}
+				clientVMI = runVMIAndExpectLaunch(clientVMI, 240)
+				clientVMIPodName = tests.GetVmPodName(virtClient, clientVMI)
+
+				Expect(console.LoginToFedora(clientVMI)).To(Succeed(), "Should be able to login to the Fedora VM")
+			})
+
+			BeforeEach(func() {
+				serverVMI = tests.NewRandomVMIWithEphemeralDisk(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling))
+				serverVMI.Spec.Domain.Resources.Requests[k8sv1.ResourceMemory] = resource.MustParse(fedoraVMSize)
+				serverVMI = runVMIAndExpectLaunch(serverVMI, 240)
+				serverIP, err = waitVMIfaceIPReport(serverVMI, defaultNetworkName, ifaceIPReportTimeout)
+				Expect(err).NotTo(HaveOccurred(), "should have managed to figure out the IP of the server VMI")
+				Expect(console.LoginToFedora(serverVMI)).To(Succeed(), "Should be able to login to the Fedora VM")
+
+				// TODO test also the IPv6 address (issue- https://github.com/kubevirt/kubevirt/issues/7506)
+				libnet.SkipWhenClusterNotSupportIpv4(virtClient)
+				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed(), "connectivity is expected *before* migrating the VMI")
+			})
+
+			It("[test_id:2656]should keep connectivity after a migration", func() {
+				migration := tests.NewRandomMigration(clientVMI.Name, clientVMI.GetNamespace())
+				_ = tests.RunMigrationAndExpectCompletion(virtClient, migration, tests.MigrationWaitTime)
+				// In case of clientVMI and serverVMI running on the same node before migration, the serverVMI
+				// will be reachable only when the original launcher pod terminates.
+				Eventually(func() error {
+					return waitForPodCompleted(clientVMI.Namespace, clientVMIPodName)
+				}, tests.ContainerCompletionWaitTime, time.Second).Should(Succeed(), fmt.Sprintf("all containers should complete in source virt-launcher pod: %s", clientVMIPodName))
+
+				Expect(libnet.PingFromVMConsole(clientVMI, serverIP)).To(Succeed(), "connectivity is expected *after* migrating the VMI")
```

Comment:
doesn't the VMI get a new IP after live migration? How is it possible that pinging the server IP which is retrieved before the migration works?